checker/jtreg/nullness/onlyannotatedfor/AnnotatedForWithUse.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * @test
+ *
+ * @summary Test that difference default applies to unannotated code.
+ * @compile/fail/ref=OnlyAnnotatedForWithUse.out -XDrawDiagnostics -Xlint:unchecked -processor org.checkerframework.checker.nullness.NullnessChecker -AonlyAnnotatedFor AnnotatedForWithUse.java
+ * @compile/fail/ref=ConservativeDefaultWithUse.out -XDrawDiagnostics -Xlint:unchecked -processor org.checkerframework.checker.nullness.NullnessChecker -AuseConservativeDefaultsForUncheckedCode=source AnnotatedForWithUse.java
+ */
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.framework.qual.AnnotatedFor;
+
+public class AnnotatedForWithUse {
+    class Unannotated {
+        Object o;
+        Object get() { return null; }
+        void set(Object of) {}
+    }
+
+    @AnnotatedFor("nullness")
+    class AnnotatedUse {
+        void use(Unannotated u) {
+            @NonNull Object obj =  u.o;
```

Comment:
Some inline comments would be useful to make it reproducible what you want to test.
Maybe introduce some short-cut notation and say what you expect for each:

````
// 1: OK, 2: Err, but want OK, 3: Err
````

or something like that, where 1, 2, 3 are the different configurations. For each, what is the expected behavior and whether it is what we actually want.
E.g. for the field update, conservative defaults don't work correctly.